updated mdx with new cpp code

ashT001 opened the pull request:
Added C++ tabs to the Starting With Sprite Tutorial and Starting With Sprite Layering Tutorial MDXs

Not perfect, needs feedback, as well as decisions regarding certain aesthetic / functional showcases (for instance: when referencing/hyperlinking a function, is it done so with the C++ or C# naming convention?)

ashT001 committed:
updated mdx with new cpp code

ashT001 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Add C++ styling to Sprite tutorial series MDX](https://trello.com/c/nQYEBnQl/570-add-c-styling-to-sprite-tutorial-series-mdx)

Liquidscroll commented on Tutorials/Sprite-Related-Tutorials/Getting Started With Sprites in Splashkit Tutorial - C#/Getting Started With Sprites in Splashkit Tutorial - CSharp.mdx:
````diff
@@ -23,47 +23,85 @@ don't interact with those.
+  <TabItem label="C#">
+  ```csharp
 Bitmap happySun;
+Sprite happySunSprite;
````
You should probably remove Sprite happySunSprite as it's created later on in step 2.

Liquidscroll commented on Tutorials/Sprite-Related-Tutorials/Getting Started With Sprites in Splashkit Tutorial - C#/Getting Started With Sprites in Splashkit Tutorial - CSharp.mdx:
````diff
@@ -23,47 +23,85 @@ don't interact with those.
+  <TabItem label="C++">
+```c++
+public static Bitmap SplashKit.CreateBitmap(string name, int width, int height);
+public Bitmap(string name, int width, int height);
````
This is C# code. I think you want: `bitmap load_bitmap(string name, string filename)` (https://splashkit.io/api/graphics/#load-bitmap) and `bitmap create_bitmap(string name, int width, int height)` (https://splashkit.io/api/graphics/#create-bitmap)

Liquidscroll commented on Tutorials/Sprite-Related-Tutorials/Getting Started With Sprites in Splashkit Tutorial - C#/Getting Started With Sprites in Splashkit Tutorial - CSharp.mdx:
````diff
@@ -23,47 +23,85 @@ don't interact with those.
+  </TabItem>
+  <TabItem label="C++">
+```c++
+    sprite happySunSprite = create_sprite("Happy Sun", HappySun);
````
Should probably change this to `sprite happySunSprite;` to maintain consistency with C# code and following step.

Liquidscroll commented on Tutorials/Sprite-Related-Tutorials/Getting Started With Sprites in Splashkit Tutorial - C#/Getting Started With Sprites in Splashkit Tutorial - CSharp.mdx:
````diff
@@ -73,10 +111,20 @@ So we have a Sprite, `happySunSprite` (or whatever you have named yours.) Let's
+  <TabItem label="C++">
+```c++
+public static Window SplashKit.OpenWindow(string caption, int width, int height);
+public Window(string caption, int width, int height);
````
C# code again, I think you want: 
```
Window spriteWindow;
spriteWindow = open_window("Learning Sprites!, 800, 800);
```
So that this matches the C# code, see here: (https://splashkit.io/api/windows/#open-window).

Liquidscroll commented on Tutorials/Sprite-Related-Tutorials/Getting Started With Sprites in Splashkit Tutorial - C#/Getting Started With Sprites in Splashkit Tutorial - CSharp.mdx:
````diff
@@ -105,21 +165,32 @@ is because we still have to do a basic
+  </TabItem>
+  <TabItem label="C++">
+```c++
+  refresh_screen();
````
We might want to pass in the target fps, like so: `refresh_screen(60);` just so this matches with the C#.

Liquidscroll commented on Tutorials/Sprite-Related-Tutorials/Getting Started With Sprites in Splashkit Tutorial - C#/Getting Started With Sprites in Splashkit Tutorial - CSharp.mdx:
```diff
@@ -154,6 +225,36 @@ namespace learningSprites
+    sprite HappySunSprite = create_sprite("Happy Sun", HappySun);
+
+    // Creation of Window
+    open_window("Learning Sprites!", 800, 800);
```
I think you should change this to assign the window to a variable, like: `Window spriteWindow = open_window("Learning Sprites!", 800, 800);`

Liquidscroll commented on Tutorials/Sprite-Related-Tutorials/Getting Started With Sprites in Splashkit Tutorial - C#/Getting Started With Sprites in Splashkit Tutorial - CSharp.mdx:
````diff
@@ -195,13 +314,26 @@ reading on ahead. What specific process are we missing that would cause _every_
+  </TabItem>
+  <TabItem label="C++">
+```c++
+clear_window(window wind, color clr)
````
This should be something like `clear_window(spriteWindow, Color.White);`. I'm not sure if the colours are defined so you might need to pass in RGB values to get the white.

Liquidscroll commented on Tutorials/Sprite-Related-Tutorials/Getting Started With Sprites in Splashkit Tutorial - C#/Getting Started With Sprites in Splashkit Tutorial - CSharp.mdx:
```diff
@@ -281,14 +431,48 @@ public class Program{
+
+    refresh_screen();
+    delay(10);
+}
```
The formatting here needs to be slightly fixed (i.e. tabs added before the curly braces in this for loop). You'll likely want to make changes mentioned earlier here too.

Liquidscroll commented on Tutorials/Sprite-Related-Tutorials/Sprite Layering Tutorial/01-02-sprites-layering.mdx:
````diff
@@ -47,56 +69,118 @@ With our code at this stage, we basically just have a single-layered `Sprite`. U
+  <TabItem label="C++">
+```c++
+SplashKit.SpriteAddLayer(bodySprite, hatBit, "hat");                        // layer 1
+SplashKit.SpriteAddLayer(bodySprite, starBit, "star");                      // layer 2
````
C# code here, you'll likely want 
```
sprite_add_layer(bodySprite, hatBit, "hat");
sprite_add_layer(bodySprite, starBit, "star");
```
See: (https://splashkit.io/api/sprites/#sprite-add-layer)

Liquidscroll commented on Tutorials/Sprite-Related-Tutorials/Sprite Layering Tutorial/01-02-sprites-layering.mdx:
````diff
@@ -47,56 +69,118 @@ With our code at this stage, we basically just have a single-layered `Sprite`. U
+  <TabItem label="C++">
+```c++
+sprite_show_layer(bodySprite, 1);
+clear_screen(COLOR_WHITE);
````
Should the screen be cleared here? It isn't in the C# code.

Alternatively you could modify the C# code to clear the screen. Not sure what the differences might be.

Liquidscroll commented on Tutorials/Sprite-Related-Tutorials/Sprite Layering Tutorial/01-02-sprites-layering.mdx:
```diff
@@ -47,56 +69,118 @@ With our code at this stage, we basically just have a single-layered `Sprite`. U
+SplashKit.ClearScreen(Color.White);
+SplashKit.DrawSprite(bodySprite);
+spriteLayers.Refresh();
+SplashKit.Delay(2000);
```
This is C# code. Also same question as above RE: clearing screen. Should we be clearing the screen? If we are then we should add it to the C# code.

Liquidscroll reviewed (changes requested):
Hi!

Good job with this! There's a few discrepancies/suggested fixes through out. In a few places you've used C# code instead of C++.

In response to your question regarding hyperlinking, I'm not sure if we have a specific way it should be done but in my opinion as long as the actual link to the function itself is correct then I would say that you should use whatever you're comfortable with. I think it's a pretty reasonable expectation for the learner/reader to familiarise themselves with the SplashKit library and the available bindings. 

Speaking of links, you may want to change the links throughout the documents to use relative paths rather than linking to the website (e.g., https://splashkit.io/api/sprites/#create-sprite could be just /api/sprites/#create-sprite ).

ashT001 committed:
general format fixes and code cleanup

ashT001 commented:
Hey, thanks a bunch for the feedback. I've gone and fixed up a fair few of the mistakes that were made, namely the wrong language, lot of back and forth switching from the splashkit.io website and my own VSC instances probably accounts for that, as well as the placeholder references from .io that I didn't pick up on when I was doing my final look.

I think that the screens need to be cleared to make way for instances where sprite layers are removed, it will work fine if all you ever do is add sprite layers, but if you remove one and don't wipe the screen with white it won't be shown as removed, so I've added that into the C# code also.

Overall really appreciative of the good feedback!

Liquidscroll reviewed (approved):
Looks good to me!
